### PR TITLE
meter request and exception rates in cassandra client pool

### PR DESCRIPTION
**Pull request by jnelson15**
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/1497)

### CODE REVIEW

**Comment by clockfort** [comment]
👍

**Comment by rhero** [comment]
@hsaraogi to review from London just so we have SA.

**Comment by jnelson15** [comment]
@rhero we good to merge? I don't have write access.

**Comment by rhero** [comment]
Merged!

**Comment by jnelson15** [comment]
Thanks!

File: atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPool.java [hunk 1]
```diff
@@ -145,27 +145,46 @@ public String toString() {
...
-        private final Counter totalRequestExceptions = new Counter();
-        private final Counter totalRequestConnectionExceptions = new Counter();
+    private class RequestMetrics {
+        private final Meter totalRequests;
+        private final Meter totalRequestExceptions;
+        private final Meter totalRequestConnectionExceptions;
 
-        Counter getTotalRequests() {
-            return totalRequests;
+        RequestMetrics() {
```
Comment (jboreiko) [inline atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPool.java]: nit: this code can be de-duped by using the below method with an empty string (the metrics registry will happily handle them) - From MetricsRegistry.java:
```
    /**
     * Concatenates a class name and elements to form a dotted name, eliding any null values or
     * empty strings.
     *
     * @param klass    the first element of the name
     * @param names    the remaining elements of the name
     * @return {@code klass} and {@code names} concatenated by periods
     */
    public static String name(Class<?> klass, String... names) {
```

File: atlasdb-client/src/main/java/com/palantir/atlasdb/util/MetricsManager.java [hunk 2]
```diff
@@ -54,6 +55,20 @@ private synchronized void registerMetricWithFqn(String fullyQualifiedMetricName,
         }
     }
 
+    public Meter registerMeter(Class clazz, String meterName) {
```
Comment (jboreiko) [inline atlasdb-client/src/main/java/com/palantir/atlasdb/util/MetricsManager.java]: Same here as above with the dedup.

File: atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPool.java (hunk 1 above)
Comment (jnelson15) [inline atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPool.java]: Nice didn't realize that! Will add a follow-up PR.

**Pull request merged**